### PR TITLE
[ACS-6842] [ACA] Clicking "Create" button multiple times in folder creation modal sends the same folder creation request multiple times

### PR DESCRIPTION
**Pull request by DaryaBalvanovich**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Create button still enable after clicking. Multiple API calls.

https://github.com/Alfresco/alfresco-ng2-components/assets/166367848/a5286810-1f51-4597-8f8a-ab17b34d1c64

**What is the new behaviour?**
Create button disable after clicking.
Also did small refactor with names and code organization (moved private functions to the end of the file, refactor some variable declarations, added subjects in tests to prevent multiple spy creation)

https://github.com/Alfresco/alfresco-ng2-components/assets/166367848/ce0c8a23-ef99-4f5c-99eb-f4cfadee2980

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/dialogs/folder.dialog.spec.ts [hunk 1]
```diff
@@ -302,4 +302,18 @@ describe('FolderDialogComponent', () => {
             });
         });
     });
+
+    it('should disable the submit button after submitting the form', () => {
```
Comment (MichalKinas) [inline lib/content-services/src/lib/dialogs/folder.dialog.spec.ts:306]: I think this test case should have an additional step to fill the name of the folder in and then perform submit, preferably by emulating the button click instead of calling the submit directly from the component. In this case we will be sure that button was enabled and then got disabled by submitting, right now the button is disabled by default when dialog is opened

Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/dialogs/folder.dialog.spec.ts]: Thanks for catching!
I refactored the tests and now we have both tests: button click and component submit call. I also reorganized the tests a bit to reduce the amount of duplicate code, please take a look

**Review by DenysVuika — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

**Review by g-jaskowski — APPROVED** [review]
